### PR TITLE
Change to json bodies on put/post

### PR DESCRIPTION
**Pull request by jwiggins78**
This change would require all users to be on version 5.0 of the API or better.  Given that this is a node library, I don't think that is a problem.

It fixes the fact that PUTs aren't working now and gets closer to a normal API pattern.  Please play around with it and let me know if you agree.

### CODE REVIEW

File: src/Api.ts [hunk 1]
```diff
@@ -381,21 +382,21 @@ export class Api {
...
         else {
-            headers.append('Content-Type', 'application/x-www-form-urlencoded')
-            bodyParams = Object.keys(params).reduce(function(a, k) {
-                a.push(k + '=' + encodeURIComponent(params[k]))
-                return a
-            }, []).join('&')
-            if (method === Api.Methods.GET) {
-                if (bodyParams) {
-                    queryString = '?' + bodyParams
+            headers.append('Content-Type', 'application/json')
```
Comment (citizensas) [inline src/Api.ts]: content-type should be `application/x-www-form-urlencoded` when the request method is not `POST` or `PUT`

**Comment by citizensas** [comment]
I've a few concerns about this PR. 
Firstly I don't know if it's a good idea to make all users to use version >5.0. I guess we could have a simple `if` statement for <5.0 to fallback to its initial code.

Have you considered instead of changing content-type to json for all users we add an option for it?
It can be a default option too.

Other than that, I'm all for having nice RESTfull github-like api, that accepts json as payload.

File: src/Api.ts (hunk 1 above)
Comment (jwiggins78) [inline src/Api.ts]: I put in a conditional switch.  Let me know if this feels a little better.

**Comment by jwiggins78** [comment]
@citizensas I agree that this would be a breaking change.  I'd suggest that we maybe version appropriately as a result.  Here are my reasons for even suggesting such a breaking change.

1. People using this library are likely using it v5.0 or better as v5.0 has been out since 2011 and this library wasn't even created until 2014.  It's time we started to push people to use more modern versions of the API.  I thought about changing the interface to take a number for the version so it's easier to validate, but decided not to touch it.  Anyway, I really, really don't think backward compatibility that far is necessary.

2. I attempted to use this library for the hack-a-thon and it currently doesn't work on edits.  You get a 200 OK back, but nothing changed.  This is why I would not support the existing code in the way suggested for versions prior to 5.0.  My previous pull request fixed how apiKeys are used, but I still have this problem.  I found stack overflow comments where people were calling request() directly to work around this problem.  That should be reason enough to fix that problem.  I had trouble fixing it using other encodings.  It's just cleaner to move to json bodies.

I'm open either way, but would really like to get the fact that edits don't work addressed as the more I start to use serverless and node.js, the more I need a solid library for communicating with redrock.  Let me know how you'd prefer to proceed and thanks for the review!

**Comment by citizensas** [comment]
@jwiggins78 I've updated a bunch of dependencies so I'd like you to update your branch from master. I've created a branch named `v3` that where the current version will live. 
I'll bump a major version after your Pull Request is merged and then will work on creating a release note.

Also I'd like you to remove `package-lock.json` file from `.gitignore`, since npm states [here](https://docs.npmjs.com/files/package-lock.json) that it needs to be in the repository.

> This file is intended to be committed into source repositories

If you don't want to delete `package-lock.json` after every build, I suggest you downgrade your npm to version 4 (this did it for me `npm i -g npm@4`).

After merging upstream and pushing your commits the CI should pass, hopefully.

**Pull request merged**